### PR TITLE
[cloud_functions] Migrate to the new embedding

### PR DESCRIPTION
**Pull request by collinjackson**
## Description

- Migrate the cloud_functions plugin to use the new embedder.
- Migrated to Firebase Test Lab compatible e2e testing.

## Related Issues

https://github.com/flutter/flutter/issues/41865

### CODE REVIEW

File: packages/cloud_functions/example/android/app/src/androidTest/java/io/flutter/plugins/firebase/cloudfunctionsexample/EmbedderV1ActivityTest.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.firebase.cloudfunctionsexample;
+\
```
Comment (xster) [inline packages/cloud_functions/example/android/app/src/androidTest/java/io/flutter/plugins/firebase/cloudfunctionsexample/EmbedderV1ActivityTest.java]: seems unintentional

File: packages/cloud_functions/example/android/app/src/androidTest/java/io/flutter/plugins/firebase/cloudfunctionsexample/MainActivityTest.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.firebase.cloudfunctionsexample;
+\
```
Comment (xster) [inline packages/cloud_functions/example/android/app/src/androidTest/java/io/flutter/plugins/firebase/cloudfunctionsexample/MainActivityTest.java]: ditto

**Comment by xster** [comment]
@collinjackson can you sync this one up again?

**Comment by xster** [comment]
gentle ping on this one

**Comment by Salakar** [comment]
Closing as no longer required, already updated. Thanks

**Pull request closed without merging**